Source/Tools/FEXServer/FDCountWatch.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+#include <sys/types.h>
+
+namespace FDCountWatch {
+  void GetMaxFDs();
+  void IncrementFDCountAndCheckLimits(ssize_t Num);
```

Comment:
Should have a comment mentioning `Num` may be negative, since the name suggests it shouldn't.

Actually might be better to just duplicate this interface, adding an explicit `DecrementFDCount` that negates its argument and forwards to a common internal-only function. This would get rid of the strangely looking `Increment...(-thing.size() - 1)` calls entirely.